### DISCUSSION TITLE
get_params option?

### DISCUSSION THREAD

**Discussion by FABoulanger** [ourownstory/neural_prophet#238] [Q&A - get help using NeuralProphet] [answered]
Hello,

do you know how to access the parameters of the models in order to have them in a dictionary?
In scikit learn there is model.get_params, is there an equivalent?
thank you

**Comment by HansikaPH** [comment, marked as answer]
Thanks for this suggestion. At the moment, there is no function accessible to the users to do this. One option would be to directly access the model parameters as in `m.model.season_params.items()` for the seasonality related parameters etc. But, having such a function to do that would be certainly useful. If you would like to give it a try in implementing it, we would be happy to help and collaborate.

**Comment by ourownstory** [comment]
@FABoulanger This is actually a great suggestion and makes for a great first issue. I added it as issue #252 in case you or someone would like to contribute it.

Further, for hyperparameters, you can retrieve them by calling `asdict` from `dataclasses`, [see on stackoverflow, Option 5](https://stackoverflow.com/questions/35282222/in-python-how-do-i-cast-a-class-object-to-a-dict/35282286), as they are all dataclasses (code [here](https://github.com/ourownstory/neural_prophet/blob/16f90a0b8dbc8f7918c9c6e4fa28a6518cf20d85/neuralprophet/configure.py#L17)).
e.g. `trend_hyperparams = dataclasses.asdict(m.config_trend)`
